### ISSUE TITLE
Regression in mypy 0.930: Cannot override writable attribute "__module__" with a final one

### ISSUE THREAD

**Issue by dvarrazzo** [python/mypy#11820]
**Bug Report**

The following code is reported as error by mypy 0.930 (Python 3.8). It seems correct, and was considered correct so far:

```python
from enum import Enum

class PyFormat(str, Enum):
    __module__ = "psycopg.adapt"
    AUTO = "s"
    TEXT = "t"
    BINARY = "b"
```

The error reported is:
```
$ mypy test_mypy.py 
test_mypy.py:4: error: Cannot override writable attribute "__module__" with a final one  [misc]
```

which I don't even know what means...

**Comment by sobolevn** [comment]
Thanks! Fix is on its way. You can add `# type: ignore` for now.

**Comment by dvarrazzo** [comment]
I'm afraid it will have to be "mypy != 0.930" because there are too many of them and because psycopg is also affected by https://github.com/python/typeshed/issues/6661 :smile: 

https://github.com/psycopg/psycopg/runs/4610348714?check_suite_focus=true

**Comment by hauntsaninja** [comment]
I'll add pyscopg to mypy_primer, so hopefully you'll have fewer regressions in the future :-)

**Comment by dvarrazzo** [comment]
@hauntsaninja amazing, thank you! And thank you all for the quick turnaround!

**Comment by dvarrazzo** [comment]
This seems still open in mypy 0.931. In which release is the fix due?

https://github.com/psycopg/psycopg/runs/4741795872?check_suite_focus=true

**Comment by hauntsaninja** [comment]
The commit made it in, but I don't think it actually fixed. I can repro on both master and 0.931, cc @sobolevn

**Comment by sobolevn** [comment]
I am so sorry 🤦 
I've literally confused `and` and `or` operations.

Tests didn't catch this, because in tests `__module__` is not defined on `object`.
Now it should be fine! I've double checked it locally.

<img width="752" alt="Снимок экрана 2022-01-09 в 16 01 55" src="https://user-images.githubusercontent.com/4660275/148683212-5cb2e952-c81e-4d1f-8f1e-dc4f9fa52527.png">

**Comment by dvarrazzo** [comment]
No worries, @sobolevn: your work is very appreciated :slightly_smiling_face:
